exporters/trace/stackdriver/src/test/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandlerProtoTest.java

```diff
@@ -258,6 +262,35 @@ public void generateSpan() {
         .isEqualTo(Int32Value.newBuilder().setValue(CHILD_SPAN_COUNT).build());
   }
 
+  private static void verifyResourceLabels(Map<String, AttributeValue> attributeMap) {
+    if (StackdriverV2ExporterHandler.RESOURCE == null) {
+      return;
+    }
+    ResourceType resourceType = StackdriverV2ExporterHandler.RESOURCE.getResourceType();
+    switch (resourceType) {
```

Comment:
It looks like it is possible for this method to not test the logic of putResourceLabels if the ResourceType isn't detected.  Is there a way to refactor the code so that the ResourceType is passed in to generateSpan, or putResourceLabels is tested directly?  Then we could easily test the code for all three cases, independent of the environment.

Reply:
Done. The constructor of `MonitoredResource` is not public so I used some mocks here.